spark3-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -71,6 +71,8 @@ statement
     | ALTER TABLE multipartIdentifier DROP PARTITION FIELD transform                        #dropPartitionField
     | ALTER TABLE multipartIdentifier REPLACE PARTITION FIELD transform WITH transform (AS name=identifier)? #replacePartitionField
     | ALTER TABLE multipartIdentifier WRITE writeSpec                                       #setWriteDistributionAndOrdering
+    | ALTER TABLE multipartIdentifier REPLACE ROW_IDENTIFIER FIELDS '(' fields+=identifier (',' fields+=identifier)* ')' #replaceIdentifierFields
```

Comment:
do we now have `SET ROW_IDENTIFIER`? if not, wondering if it would seem odd if user can only "replace" even when the table doesn't have row identifiers yet

Reply:
+1 for @yyanyy 's comment.

Reply:
yeah `SET` makes sense to me. Was trying to reuse `REPLACE` keyword, but it does feel awkward.